tests/csm/rest/test_iam_users.py

```diff
@@ -424,24 +437,234 @@ def test_35930(self):
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.s3_ops
+    @pytest.mark.tags('TEST-39411')
+    def test_39411(self):
+        """
+        S3 IAM User: Create same name buckets with IAM users of same name in different tenant
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info(
+            "[START] Create same name buckets with same name IAM users in different tenant")
+        for cnt in range(2):
+            tenant = "tenant_" + str(cnt)
+            self.log.info("Creating new iam user with tenant %s", tenant)
+            optional_payload = self.csm_obj.iam_user_payload_rgw("loaded")
+            optional_payload.update({"tenant": tenant})
+            optional_payload.update({"display_name": self.display_name})
+            self.log.info("updated payload :  %s", optional_payload)
+            resp1 = self.csm_obj.create_iam_user_rgw(optional_payload)
+            self.log.info("Verify Response : %s", resp1)
+            assert_utils.assert_true(resp1.status_code == HTTPStatus.CREATED,
+                                     "IAM user creation failed")
+            self.created_iam_users.add(resp1.json()['tenant'] + "$" + optional_payload['uid'])
+            resp = self.csm_obj.compare_iam_payload_response(resp1, optional_payload)
+            self.log.info("Printing response %s", resp)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.log.info("Step: Verify no bucket present in new account")
+            s3_obj = S3TestLib(access_key=resp1.json()["keys"][0]["access_key"],
+                               secret_key=resp1.json()["keys"][0]["secret_key"])
+            buckets = s3_obj.bucket_list()[1]
+            assert_utils.assert_false(len(buckets), "buckets found on new IAM user")
+            self.log.info("Step: Verified no bucket present in new account")
+            self.log.info("Create bucket and perform IO")
+            self.log.info("Step: Verify create bucket")
+            status, resp = s3_obj.create_bucket(self.bucket_name)
+            assert_utils.assert_true(status, resp)
+            resp = s3_obj.bucket_list()
+            assert_utils.assert_in(self.bucket_name, resp[1], resp[1])
+            self.log.info("Step: Verified create bucket")
+            self.log.info("Step: Check bucket is empty")
+            resp = s3_obj.object_list(self.bucket_name)
+            assert resp[0], resp[1]
+            assert_utils.assert_true(resp[0], resp[1])
+            resp_bkt_lst = None if not resp[1] else resp[1]
+            # For checking the object list should be none
+            assert resp_bkt_lst is None, resp
+            assert_utils.assert_is_not_none(resp[0], resp[1])
+            self.log.info("Step: Verified that bucket was empty")
+            self.log.info("Step: Verify put object.")
+            test_file = "test-object.txt"
```

Comment:
I can see repeated and duplicate code from 561 to 572. It would be better to move section to setup and teardown method. Please, refer s3 tests.

Reply:
this is in mid of the test, after creating bucket
due to a product bug, have added this check

Reply:
@DhobaleVishal @RajeshPratapSInghChouhan agree with Vishal's suggestion that code duplication has to be reduced and I see Rajesh's point that this cannot be taken into account here. I have added an alternate suggestion that we can re-purpose s3_common_test_lib's create_bucket_put_object to reduce code duplication. Rajesh can update create_bucket_put_object to add an additional step to verify that the bucket created is empty

Reply:
if we use create_bucket_put_object, will not be able to verify if the bucket created was empty,
will add this check into the lib function then